actions/appSettings.ts

```diff
@@ -1,58 +1,81 @@
 'use server';
 
 import { redirect } from 'next/navigation';
+import { type z } from 'zod';
 import { safeRevalidateTag } from '~/lib/cache';
+import { appSettingSchema } from '~/schemas/appSettings';
 import { createEnvironmentFormSchema } from '~/schemas/environment';
 import { requireApiAuth } from '~/utils/auth';
 import { prisma } from '~/utils/db';
 
-export async function setAnonymousRecruitment(input: boolean) {
-  await requireApiAuth();
+// Generic function to set an app setting with validation
+export async function setAppSetting<
+  Key extends keyof z.infer<typeof appSettingSchema>,
+>(key: Key, value: z.infer<typeof appSettingSchema>[Key]) {
+  console.log('setting app setting', key, value);
+
```

Comment:
I think this needs an auth check, right?

Reply:
resolved [b9da84f](https://github.com/complexdatacollective/Fresco/pull/161/commits/b9da84f06fc178d6c9906a2aabf7e3958146c5d7)